File: leetcode-submissions/0053-maximum-subarray/solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:

        # DP: O(n) time and constant space
        n = len(nums)
        curr_sum = nums[0]
        maxx = curr_sum

        for i in range(1, n):
            curr_sum = max(curr_sum + nums[i], nums[i])
            maxx = max(maxx, curr_sum)

        return maxx

File: leetcode-submissions/0053-maximum-subarray/solution.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) 
    {
        int sum = 0;
        int best = INT_MIN;
        int length = nums.size();

        for(int i = 0; i < length; i++)
        {
            sum = max(nums[i], sum + nums[i]);
            best = max(best, sum);
        }
        return best;
    }
};
